File: dotnet/src/MyDotey.SCF.Simple/Source/StringProperty/PropertieFile/PropertiesFileConfigurationSourceConfig.cs
using System;

namespace MyDotey.SCF.Source.StringProperty.PropertiesFile
{
    /**
     * @author koqizhao
     *
     * May 17, 2018
     */
    public class PropertiesFileConfigurationSourceConfig : DefaultConfigurationSourceConfig
    {
        public virtual String FileName { get; private set; }

        protected PropertiesFileConfigurationSourceConfig()
        {

        }

        public override String ToString()
        {
            return String.Format("{0} {{ name: {1}, fileName: {2} }}", GetType().Name, Name, FileName);
        }

        public new class Builder
            : DefaultConfigurationSourceConfig.DefaultAbstractBuilder<Builder, PropertiesFileConfigurationSourceConfig>
        {
            protected override PropertiesFileConfigurationSourceConfig NewConfig()
            {
                return new PropertiesFileConfigurationSourceConfig();
            }

            public virtual Builder SetFileName(String fileName)
            {
                Config.FileName = fileName;
                return this;
            }

            public override PropertiesFileConfigurationSourceConfig Build()
            {
                if (string.IsNullOrWhiteSpace(Config.FileName))
                    throw new ArgumentNullException("fileName is null or empty");

                Config.FileName = Config.FileName.Trim();
                if (!Config.FileName.EndsWith(".properties"))
                    Config.FileName += ".properties";

                return base.Build();
            }
        }
    }
}

File: java/src/main/java/org/mydotey/scf/source/stringproperty/propertiesfile/PropertiesFileConfigurationSourceConfig.java
package org.mydotey.scf.source.stringproperty.propertiesfile;

import org.mydotey.scf.DefaultConfigurationSourceConfig;
import org.mydotey.scf.source.FileSourceType;

/**
 * @author koqizhao
 *
 * May 17, 2018
 */
public class PropertiesFileConfigurationSourceConfig extends DefaultConfigurationSourceConfig {

    private String _fileName;
    private FileSourceType _type;

    protected PropertiesFileConfigurationSourceConfig() {

    }

    public String getFileName() {
        return _fileName;
    }

    public FileSourceType getType() {
        return _type;
    }

    @Override
    public String toString() {
        return String.format("%s { name: %s, fileName: %s, type: %s }", getClass().getSimpleName(), getName(),
            getFileName(), getType());
    }

    public static class Builder extends
            DefaultConfigurationSourceConfig.DefaultAbstractBuilder<Builder, PropertiesFileConfigurationSourceConfig> {

        @Override
        protected PropertiesFileConfigurationSourceConfig newConfig() {
            return new PropertiesFileConfigurationSourceConfig();
        }

        public Builder setFileName(String fileName) {
            getConfig()._fileName = fileName;
            return this;
        }

        public Builder setType(FileSourceType type) {
            getConfig()._type = type;
            return this;
        }

        @Override
        public PropertiesFileConfigurationSourceConfig build() {
            if (getConfig().getFileName() == null || getConfig().getFileName().trim().isEmpty())
                throw new IllegalArgumentException("fileName is null or empty");

            getConfig()._fileName = getConfig()._fileName.trim();
            if (!getConfig()._fileName.endsWith(".properties"))
                getConfig()._fileName += ".properties";

            if (getConfig().getType() == null)
                getConfig()._type = FileSourceType.CLASS_PATH;

            return super.build();
        }

    }

}
